File: 7kyu 2/longest.js
/*
|--------------------------------------------------------------------------
| Two to One
|--------------------------------------------------------------------------
|
*/

const longest = (s1, s2) => [...new Set([...new Set(s1)].concat([...new Set(s2)]))].sort().join("");


const longest = (s1, s2) => [...new Set(s1 + s2)].sort().join('');


function longest(s1, s2) {
   return Array.from(new Set(s1 + s2)).sort().join('');
}


const longest = (s1, s2) => Array.from(s1 + s2).sort().filter((x, i, a) => x !== a[i - 1]).join('');


console.log(longest("aretheyhere", "yestheyarehere"), "aehrsty")
console.log(longest("loopingisfunbutdangerous", "lessdangerousthancoding"), "abcdefghilnoprstu")
console.log(longest("inmanylanguages", "theresapairoffunctions"), "acefghilmnoprstuy")

File: Python/7py 2/longest.py
"""
|--------------------------------------------------------------------------
| Two to One
|--------------------------------------------------------------------------
|
"""

def longest(a1, a2):
   return ''.join(sorted(set(a1+a2)))


def longest(s1, s2):
   longest = ""
   for i in s1: 
      if i not in longest:
         longest += i
   for i in s2:
      if i not in longest:
         longest += i 
   return ''.join(sorted(longest))


def longest(s1, s2):
   return ''.join(sorted(set(s1).union(s2)))


def longest(s1, s2):
   return "".join([x for x in "abcdefghijklmnopqrstuvwxyz" if x in s1+s2])


print(longest("aretheyhere", "yestheyarehere"), "aehrsty")
print(longest("loopingisfunbutdangerous", "lessdangerousthancoding"), "abcdefghilnoprstu")
print(longest("inmanylanguages", "theresapairoffunctions"), "acefghilmnoprstuy")